gammapy/estimators/utils.py

```diff
@@ -770,83 +773,350 @@ def combine_flux_maps(maps, method="gaussian_errors", reference_model=None):
+            mask = mask_valid & mask_valid_k
+            mask_k = ~mask_valid & mask_valid_k
+
+            mean.data[mask] = (
+                (mean.data * sigma_k**2 + mean_k * sigma.data**2)
+                / (sigma.data**2 + sigma_k**2)
+            )[mask]
+            sigma.data[mask] = (
+                sigma.data * sigma_k / np.sqrt(sigma.data**2 + sigma_k**2)
+            )[mask]
+
+            mean.data[mask_k] = mean_k[mask_k]
+            sigma.data[mask_k] = sigma_k[mask_k]
+
+        ts = mean * mean / sigma / sigma + ts_diff
+
+        kwargs = dict(
+            sed_type="dnde", reference_model=reference_model, meta=meta, gti=gti
+        )
+        return FluxMaps.from_maps(dict(dnde=mean, dnde_err=sigma, ts=ts), **kwargs)
+
+    elif method in ["distrib", "profile"]:
+        if dnde_scan_axis is None:
+            dnde_scan_axis = _default_scan_map(maps[0]).geom.axes["dnde"]
+        for k, map_ in enumerate(maps):
+            map_stat_scan = (
+                interpolate_profile_map(map_, dnde_scan_axis=dnde_scan_axis)
+                if method == "profile"
+                else approximate_profile_map(map_, dnde_scan_axis=dnde_scan_axis)
+            )
+            map_stat_scan.data[np.isnan(map_stat_scan.data)] = 0.0
+            if k == 0:
+                stat_scan = map_stat_scan
+            else:
+                stat_scan += map_stat_scan
+
+        return get_flux_map_from_profile(
+            {"stat_scan": stat_scan},
+            reference_model=reference_model,
+            meta=meta,
+            gti=gti,
+        )
+
+    else:
+        raise ValueError(
+            f'Invalid method provided : {method}. Available methods are : "gaussian_errors", "distrib", "profile"'
+        )
+
+
+def _default_scan_map(flux_map, dnde_scan_axis=None):
+    if dnde_scan_axis is None:
+        dnde_scan_axis = MapAxis(
+            _generate_scan_values() * flux_map.dnde_ref.squeeze(),
+            interp="lin",
+            node_type="center",
+            name="dnde",
+            unit=flux_map.dnde_ref.unit,
+        )
+
+    geom = flux_map.dnde.geom
+    geom_scan = geom.to_image().to_cube([dnde_scan_axis] + list(geom.axes))
+    return Map.from_geom(geom_scan, data=np.nan, unit="")
+
 
+def interpolate_profile_map(flux_map, dnde_scan_axis=None):
+    """Interpolate sparse likelihood profile to regular grid
+
+    Parameters
+    ----------
+    flux_map : `~gammapy.estimators.FluxMaps`
+        Flux map.
+    dnde_scan_axis : `~gammapy.maps.MapAxis`
+        dnde axis used to compute the profile.
+
+    Returns
+    -------
+    scan_map: `~gammapy.estimators.Maps`
+        Likelihood profile map.
+
+    """
+
+    stat_scan = _default_scan_map(flux_map, dnde_scan_axis)
+    dnde_scan_axis = stat_scan.geom.axes["dnde"]
+
+    mask_valid = ~np.isnan(flux_map.dnde.data)
+    dnde_scan_values = flux_map.dnde_scan_values.quantity.to_value(dnde_scan_axis.unit)
+
+    for ij, il, ik in zip(*np.where(mask_valid)):
```

Comment:
A very naive question: is it always of dimension 3?

Reply:
I don't think a time axis is supported for ExcessMap/TSmap.

Reply:
And the event types?

Reply:
for now each event type will be a dataset